### ISSUE TITLE
Figure out what to do with multiple parameter lists (esp. implicits) in Generic

### ISSUE THREAD

**Issue by joroKr21** [milessabin/shapeless#790]
```scala
case class Ord1[A: Ordering](value: A)
class Ord2[A: Ordering](val value: A)
Generic[Ord1[Int]] // works
Generic[Ord2[Int]] // doesn't work
```

**Comment by milessabin** [comment]
Good catch.

**Comment by joroKr21** [comment]
This is deeper than I thought. Implicit parameters of case class are captured at the materialization site of `Generic`. To capture them at call sites of `from` they have to become part of the `Repr`, i.e. the issue boils down to dealing with multiple parameter lists.

**Comment by joroKr21** [comment]
Fixed in #1272
